src/stacked_borrows.rs

```diff
@@ -661,14 +661,14 @@ impl<'a, 'mir, 'tcx> EvalContextExt<'tcx> for MiriEvalContext<'a, 'mir, 'tcx> {
             // Primitives of reference type, that is the one thing we are interested in.
             fn visit_primitive(&mut self, place: MPlaceTy<'tcx, Borrow>) -> EvalResult<'tcx>
             {
-                match place.layout.ty.sty {
-                    ty::Ref(_, _, mutbl) => {
-                        let val = self.ecx.read_immediate(place.into())?;
-                        let val = self.ecx.retag_reference(val, mutbl)?;
-                        self.ecx.write_immediate(val, place.into())?;
-                    }
-                    _ => {}, // nothing to do
-                }
+                let mutbl = match place.layout.ty.sty {
```

Comment:
I think this can be simplified to an if let on `builtin_deref(false)`

Reply:
Unfortunately that reports *immutable* for `Box`, making it useless.

Reply:
I added a comment explaining this.

Reply:
Wat. I need to review all uses of that method now -.-

Reply:
Please do.^^ And/or fix that method? ;)